java/arcs/core/storage/Reference.kt

```diff
@@ -31,13 +34,25 @@ data class Reference(
     val storageKey: StorageKey,
     val version: VersionMap?,
     /** Reference creation time (in milliseconds). */
-    override val creationTimestamp: Long = RawEntity.UNINITIALIZED_TIMESTAMP,
+    private var creationTimestamp_: Long = RawEntity.UNINITIALIZED_TIMESTAMP,
     /** Reference expiration time (in milliseconds). */
-    override val expirationTimestamp: Long = RawEntity.UNINITIALIZED_TIMESTAMP
+    private var expirationTimestamp_: Long = RawEntity.UNINITIALIZED_TIMESTAMP
 ) : Referencable, arcs.core.data.Reference<RawEntity> {
     /* internal */
     var dereferencer: Dereferencer<RawEntity>? = null
 
+    override val creationTimestamp: Long get() = creationTimestamp_
+    override val expirationTimestamp: Long get() = expirationTimestamp_
+
+    fun ensureTimestamps(time: Time, ttl: Ttl) {
```

Comment:
ensure timestamps are... what? This method name is a little strange. Here's a few other suggestions:

ensureTimestampsAreSet
initializeTimestamps
setTimestamps